src/query/procedure/py_module.cpp

```diff
@@ -1684,11 +1690,17 @@ PyObject *PyVertexLabelAt(PyVertex *self, PyObject *args) {
   if (!PyArg_ParseTuple(args, "n", &id)) {
     return nullptr;
   }
-  mgp_label label{nullptr};
-  if (RaiseExceptionFromErrorCode(mgp_vertex_label_at(self->vertex, id, &label))) {
+  const char *label_name{nullptr};
+  if (RaiseExceptionFromErrorCode(mgp_vertex_label_at(self->vertex, id, &label_name))) {
+    return nullptr;
+  }
+
+  auto *py_label_name = PyUnicode_FromString(label_name);
+  if (!py_label_name) {
+    PyErr_SetString(PyExc_RuntimeError, "Unable to get name from edge_type");
     return nullptr;
   }
-  return PyUnicode_FromString(label.name);
+  return py_label_name;
```

Comment:
How do these changes fix the memory corruption? What I get is we return a different error (`"Unable to get name from edge_type"`), but I don't see anything that actually fixes any memory issues.